template.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+class StringBuffer {
+  constructor() {
+    this.buffer = [];
+    this.index = 0;
+  }
+  add(str) {
+    this.buffer[this.index] = str;
+    this.index++;
+  }
+  createStr(str) {
```

Comment:
You expect that `this.index == 0` when calling `createStr` method, right? In this case, it's better to have it as a constructor, rather than a method because it should not be possible to call this method on non-empty StringBuffer.